### ISSUE TITLE
[a11y]: ContainedList [with expandable search] - keyboard action need to close expanded search

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13943]
### Package
@carbon/react

### Browser
FF

### Package version
carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

JAWS

### Description

Search edit field

Unable to close the search edit field using keyboard

### WCAG 2.1 Violation
n/a

### Reproduction/example

Record `Number 31` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

The search edit field closes When the user clicks any ware in the screen. However there is no equivalent action for the keyboard only users.

When the standard keyboard operation method is not supported, it impacts the user’s ability to interact with page controls. This affects screen reader and keyboard-only users. 

### Recommended fix: 
User should be able to close the search field using keyboard
We can use the `esc` key to close expanded search

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
